yt/data_objects/tests/test_time_series.py

```diff
@@ -14,14 +15,50 @@ def test_pattern_expansion():
             (Path(tmpdir) / file).touch()
 
         pattern = os.path.join(tmpdir, "fake_data_file_*")
-        found = get_filenames_from_glob_pattern(pattern)
+        found = DatasetSeries._get_filenames_from_glob_pattern(pattern)
         assert found == [os.path.join(tmpdir, file) for file in file_list]
 
-        found2 = get_filenames_from_glob_pattern(Path(pattern))
+        found2 = DatasetSeries._get_filenames_from_glob_pattern(Path(pattern))
         assert found2 == found
 
 
 def test_no_match_pattern():
     with tempfile.TemporaryDirectory() as tmpdir:
         pattern = os.path.join(tmpdir, "fake_data_file_*")
-        assert_raises(OSError, get_filenames_from_glob_pattern, pattern)
+        assert_raises(
+            FileNotFoundError, DatasetSeries._get_filenames_from_glob_pattern, pattern
+        )
+
+
+def test_init_fake_dataseries():
+
+    file_list = [f"fake_data_file_{str(i).zfill(4)}" for i in range(10)]
+    with tempfile.TemporaryDirectory() as tmpdir:
+        pfile_list = [Path(tmpdir) / file for file in file_list]
+        sfile_list = [os.path.join(tmpdir, f) for f in file_list]
```

Comment:
For the sake of consistency
```suggestion
        tmpdir_path = Path(tmpdir)
        pfile_list = [tmpdir_path / file for file in file_list]
        sfile_list = [os.path.join(tmpdir, file) for file in file_list]
```

Reply:
Looks good. might as well skip `os.path` and use only pathlib while we're at it